### DISCUSSION TITLE
dagrun.py ERROR - Failed to get task for ti <TaskInstance: Grp1.Grp2.name manual__... [None]>. Marking it as removed.

### DISCUSSION THREAD

**Discussion by a-meledin** [apache/airflow#34687] [Q&A]
Hello,

I have airflow 2.7.1 both locally (python 3.10) for development and in docker (python 3.8).

I've written a DAG generator function, tested it in IDE locally with dag.test() option and then put under docker run airflow instance.  Each task calls the same custom operator, with just one line logging.info() and nothing else.

When run under docker, there is an issue appeared (see title) causing last 4 tasks to be not executed at all (see image).
![image](https://github.com/apache/airflow/assets/98175818/232de057-e472-42db-a7ef-6e0dd21060d0)

Logs info:
1. Worker logs: Just error about inability to access logs for tasks marked as removed.
2. Scheduler logs: No errors
3. Docker logs: serve_logs.py: WARNING - The signature of the request was wrong
jwt.exceptions.InvalidSignatureError: Signature verification failed

Can you help where to look to investigate this issue deeper? 

Additional info:
1. It seems not connected with Custom operator, because the same behavior is for EmptyOperator.
2. It seems the issue is with Task Group generation, because when I removed them in generation, everything works
![image](https://github.com/apache/airflow/assets/98175818/a569d4bd-1f46-45e4-9cba-9580e007bad6)
For task generation I used:
- a.  For first level task group: task_group = TaskGroup(<name>, dag=adag)
- b.  For subsequent levels: task_group = TaskGroup(<name>, parent_group=parent_task_group, dag=adag)
- c.  And for task assignment: taskOp = ssh.OPJobExecOverSSH(task_id=<name>,dag=adag, task_group=parent_task_group, ....)
Is this OK?
3. Tested this DAG Generation under other Airflow 2.7.1 instance running on Pyhon 3.10.11, and encounter no such issue.... 

Thank you!

**Comment by Taragolis** [comment]
> Marking it as removed.

Task marked as removed if it doesn't persists in DAG structure anymore. Since you create it by dynamically manner I assume that you dynamic DAG generation might remove/rename tasks. 

Solution: Keep your DAG structure more stable, see [Dynamic DAGs](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dags.html#dynamic-dags)

**Reply by a-meledin** [reply]
Currently, it seems the problem is with docker image running airflow on python 3.8.18. On PyPip instances with Local executor running on Python 3.10.11 everything works normally.

**Reply by Taragolis** [reply]
Difficult to say without have reproducible examples
